src/impl/source/profiles/profileActions.ts

```diff
@@ -1,103 +1,87 @@
-            this.profileRetriever = new ProfileRetriever(org.getConnection(), debugFlag);
+            this.profileRetriever = new ProfileRetriever(org.getConnection());
         }
-        this.debugFlag = debugFlag;
     }
 
-    protected async getProfileFullNamesWithLocalStatus(profileNames: string[]): Promise<{
-        added: string[];
-        deleted: string[];
-        updated: string[];
-    }> {
-        let profilesStatus = {
-            added: [],
-            deleted: [],
-            updated: [],
-        };
-        let metadataFiles = METADATA_INFO.Profile.files || [];
-
-        //generate path for new profiles
-        let profilePath = path.join(process.cwd(), await Sfpowerkit.getDefaultFolder(), 'main', 'default', 'profiles');
-        if (metadataFiles && metadataFiles.length > 0) {
-            profilePath = path.dirname(metadataFiles[0]);
-        } else {
-            //create folder structure
-            FileUtils.mkDirByPathSync(profilePath);
-        }
+    protected async getRemoteProfilesWithLocalStatus(
+        profileNames: string[],
+        packageDirectories?: string[]
+    ): Promise<ProfileStatus> {
+        let profilesStatus: ProfileStatus = {} as ProfileStatus;
+        profilesStatus.added = [];
+        profilesStatus.updated = [];
+        profilesStatus.deleted = [];
+
+        //Load all local profiles
+        let localProfiles = await this.loadProfileFromPackageDirectories(packageDirectories);
+
+        //generate default path for new profiles
+        let profilePath = path.join(await Sfpowerkit.getDefaultFolder(), 'main', 'default', 'profiles');
+        //create folder structure
+        FileUtils.mkDirByPathSync(profilePath);
 
         // Query the profiles from org
-        const profiles = await retrieveMetadata([{ type: 'Profile', folder: null }], this.conn);
+        const remoteProfiles = await retrieveMetadata([{ type: 'Profile', folder: null }], this.conn);
 
         if (profileNames && profileNames.length > 0) {
             for (let i = 0; i < profileNames.length; i++) {
                 let profileName = profileNames[i];
                 let found = false;
 
-                for (let j = 0; j < metadataFiles.length; j++) {
-                    let profileComponent = metadataFiles[j];
-                    let retrievedProfileFileName = path.basename(
-                        profileComponent,
-                        METADATA_INFO.Profile.sourceExtension
-                    );
-                    if (profileName === retrievedProfileFileName && profiles.includes(profileName)) {
-                        profilesStatus.updated.push(profileComponent);
+                for (let j = 0; j < localProfiles.length; j++) {
+                    if (profileName === localProfiles[j].name && remoteProfiles.includes(profileName)) {
+                        profilesStatus.updated.push(localProfiles[j]);
                         found = true;
-                        break;
```

Comment:
Is it ok to remove this line?

Reply:
yeah, this is for a use case where you have the same profile (reconciled ) ones across different packages.

Reply:
> Hi @azlam-abdulsalam I tested the command and it looks good. I added one comment on one file.
> 
> Would it be possible to use the @salesforce.source-deploy-retrieve package to completely remove dependecy on this file resources\metadatainfo.json?
> 
> Thanks

I think its possible to completely remove medatainfo.json, but there are other areas such as diff, that need to be reworked